### DISCUSSION TITLE
Filament v3

### DISCUSSION THREAD

**Discussion by glennjacobs** [lunarphp/lunar#1142] [Roadmap]
Filament v2 didn't work for us for many reasons. With Livewire v3 being released soon, we decided to review the situation with Filament, specifically their v3. We felt the effort to upgrade the existing hub to Livewire v3 and of course, finish up the remaining hub functionality, might be better spent implementing Filament instead.

The changes in Filament v3 mean that it has become a true option now for Lunar. So I opened discussions with Dan Harrin (the creator of Filament). He's very keen for us to adopt Filament and promised to support us, as we will undoubtedly encounter some challenges along the way.

What this means is, when Filament v3 comes out in beta we will be working on a proof of concept for a new version of the hub. We have a long list of things we need to solve, but with Dan's help, I'm confident we can make it happen.

We understand that the current hub is being used, so we will continue to support that for a reasonable time if/when we do move to Filament v3.

With Filament powering the admin hub, it should free us up to focus on e-commerce functionality more.

### When will it be ready to use?
We have a lot to recreate, but we hope to get an MVP out ASAP.

### Can I contribute to the development?
Once we have the MVP out, we will gladly accept contributions from the community.

### How long will the old admin hub be supported?
We are unsure currently. What we do know is that once we have moved to Filament, it's unlikely the old admin hub will receive new functionality.

**Comment by adevade** [comment]
This seems like a really good move! 🥳 Excited to follow and contribute to this project!

**Comment by dan-pascu** [comment]
Looking forward for the MVP, I think it's good to move to Filament3.

**Comment by shankhadevpadam** [comment]
What is the status? I am really excited to use with filament

**Comment by gavinhewitt** [comment]
Glad to see you make this move!

**Comment by gavinhewitt** [comment]
Is there a rough ETA? Just to have an idea of what to expect.

**Comment by glennjacobs** [comment]
The Filament admin panel development is now available on the ’1.x’ branch. 

If you check the GitHub project roadmap you’ll see our outstanding tasks. 

Documentation is at very early stages, so if you want to try it you’re on your own for the moment 😆

Some sections are in, but incomplete, eg. Product Options under Settings. Please check the roadmap before submitting any bug reports.

**Reply by gavinhewitt** [reply]
Awesome guys, thank you for your hard work. Can't wait to try it out! 😍

**Comment by antal-levente** [comment]
Hello,

I'm about to start a new webshop project with Lunar and the Filament backend is very welcomed and its a very good idea (in my opinion).

I would like to ask when 1.x comes out it will be possible to upgrade from 0.8 to 1.x right?

**Reply by glennjacobs** [reply]
Yeah, there will be an automatic upgrade.

**Comment by antal-levente** [comment]
Hello,

Any updates on the progress?

Thanks,

**Reply by glennjacobs** [reply]
Yes. If you check v1 you'll see the Filament admin panel is working well in alpha. See https://docs-v1.lunarphp.io/

We are currently working on a push to get into beta.

**Comment by virtruvio** [comment]
QUESTION: With the adoption of Filament/Livewire 3, are there any considerations that are immediately-obvious to core developers and contributors for Lunar 1.x installing/coexisting with an existing Filament 3 admin panel.  I'm working on such a site for which we're considering Lunar as an e-commerce engine.

**Comment by imknight** [comment]
if I have a filament project how do I add lunar into it and allow member within my filament project to place order?

**Reply by glennjacobs** [reply]
Lunar isn't designed to be incorporated into a pre-existing panel, it provides its own panel. If you want to try add it into your own panel then you'd have to use the Resources etc that Lunar's panel does, but you may come across issues due to not using the Staff model and associated permissions. So it's not recommended.

**Reply by imknight** [reply]
So, would it be better to do it another way, using the Lunar Admin Panel and migrating whatever I have built to the Lunar Admin Panel ?

**Reply by glennjacobs** [reply]
If you'd like everything together in one panel, then the simplest thing would be to use Lunar's panel and then add in your additional Filament resources etc.

**Comment by altugsogutoglu** [comment]
Hey @glennjacobs did the migration to filament v3 succeed?
